jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -7129,5 +7129,89 @@ public boolean hasNullMarker(Field f) {
 	public int nullMarkerOffset(Field f) {
 		throw new Error("jdk.internal.misc.Unsafe.nullMarkerOffset unimplemented"); //$NON-NLS-1$
 	}
+
+	public int arrayLayout(Class<?> arrayClass) {
+		throw new Error("jdk.internal.misc.Unsafe.arrayLayout unimplemented"); //$NON-NLS-1$
+	}
+
+	public int fieldLayout(Field f) {
+		throw new Error("jdk.internal.misc.Unsafe.fieldLayout unimplemented"); //$NON-NLS-1$
+	}
+
+	public <V> V getFlatValue(Object obj, long offset, int layoutKind, Class<?> valueType) {
+		throw new Error("jdk.internal.misc.Unsafe.getFlatValue unimplemented"); //$NON-NLS-1$
+	}
+
+	public <V> void putFlatValue(Object obj, long offset, int layoutKind, Class<?> valueType, V v) {
+		throw new Error("jdk.internal.misc.Unsafe.putFlatValue unimplemented"); //$NON-NLS-1$
+	}
+
+	public final <V> Object getFlatValueVolatile(Object base, long offset, int layout, Class<?> valueType) {
```

Comment:
Why are the four additions above not also `final`?

Reply:
Actually, the class is `final`, so the `final` modifier on any method is redundant.

Reply:
I removed the finals.